modules/configuration/ajax/process.php

```diff
@@ -18,32 +22,39 @@
 require_once 'NDB_Client.class.inc';
 require_once "Utility.class.inc";
 
-$user =& User::singleton();
-if (!$user->hasPermission('config')) {
+if (!\User::singleton()->hasPermission('config')) {
     header("HTTP/1.1 403 Forbidden");
-    exit;
+    return;
 }
 
 $client = new NDB_Client();
 $client->makeCommandLine();
 $client->initialize();
 $factory = \NDB_Factory::singleton();
 $DB      = $factory->database();
+$pathIDs = getPathIDs();
 foreach ($_POST as $key => $value) {
     if (is_numeric($key)) { //update
         if ($value == "") {
-            $DB->delete('Config', array('ID' => $key));
+            $DB->update('Config', array('Value' => null), array('ID' => $key));
```

Comment:
Why is this changed?

Reply:
The basic idea of my changes is to verify whether the incoming data is a `web_path` type in the DB and if so make sure that it's a valid path.

When debugging a problem another reviewer had, i realized that the script was actually deleting values when they are blank.

This is a problem because we decided in a meeting at some point that the paths should be left blank when they're not configured (rather than allowing for `/INVALID/PLACEHOLDER/PATHS` as a value).

I figured it would make sense to keep the IDs in the database and set them to null rather than delete them. It makes more sense to me intuitively to have a setting be blank/uninitialized than for it to vanish.

Do you see this causing problems?

Reply:
Yes, when new config settings are added/unchanged from the default they do not have any entry in the Config table. The update statement will return 0 rows and report success but silently do nothing. As a result, no one will ever be able to set a config setting for the first time for any config that doesn't have a non-empty default. Not having a value in the table when the setting is blank is correct behaviour.

Reply:
> when new config settings are added/unchanged from the default they do not have any entry in the Config table

Aren't defaults stored in the Config table? ConfigSettings doesn't have any information about the value.